sqlite/tests/concurrency.rs

```diff
@@ -0,0 +1,67 @@
+use std::thread;
+use taskchampion_sync_server_core::{Storage, NIL_VERSION_ID};
+use taskchampion_sync_server_storage_sqlite::SqliteStorage;
+use tempfile::TempDir;
+use uuid::Uuid;
+
+#[test]
+fn add_version_concurrency() -> anyhow::Result<()> {
+    let tmp_dir = TempDir::new()?;
+    let client_id = Uuid::new_v4();
+
+    {
+        let con = SqliteStorage::new(tmp_dir.path())?;
+        let mut txn = con.txn()?;
+        txn.new_client(client_id, NIL_VERSION_ID)?;
+        txn.commit()?;
+    }
+
+    const N: i32 = 100;
+    const T: i32 = 4;
+
+    // Add N versions to the DB.
+    let add_versions = || {
+        let con = SqliteStorage::new(tmp_dir.path())?;
+
+        for _ in 0..N {
+            let mut txn = con.txn()?;
+            let client = txn.get_client(client_id)?.unwrap();
+            let version_id = Uuid::new_v4();
+            let parent_version_id = client.latest_version_id;
+            std::thread::yield_now(); // Make failure more likely.
+            txn.add_version(client_id, version_id, parent_version_id, b"data".to_vec())?;
+            txn.commit()?;
+        }
+
+        Ok::<_, anyhow::Error>(())
+    };
+
+    thread::scope(|s| {
+        // Spawn T threads.
+        for _ in 0..T {
+            s.spawn(add_versions);
+        }
+    });
+
+    // There should now be precisely N*T versions. This number will be smaller if there was a
+    // concurrency error allowing two conflicting `add_version` calls to overlap.
```

Comment:
```suggestion
    // There should now be precisely N*T versions. This number will be smaller if there were
    // concurrent transactions, which would have allowed two `add_version` calls with the same
    // `parent_version_id`.
```

This comment could be a little more precise about what would cause this test to fail. Specifically:

- "concurrency error" -- the point of this test is to make sure there can be *no concurrency* of add_version, whereas I think this term suggests that concurrency is supported but add_version should somehow avoid conflicts
- "overlap" -- slightly vague